File: benchmarks/reference/rust/redundant_stress.rs
fn main() {
    let mut sum: i64 = 0;
    for i in 0..500000_i64 {
        let k1 = 10 + 20;
        let k2 = k1 * 3;
        let k3 = k2 - 40;

        let a1 = i + 7;
        let a2 = a1 * 3;

        let b1 = i + 7;
        let b2 = b1 * 3;

        let c1 = 100 + 200;
        let c2 = c1 * 2;
        let c3 = c2 - 100;

        let d1 = 5 + 10;
        let d2 = d1 * 5;
        let d3 = d2 - 5;

        let live = a2 + b2 + k3 + c3 + d3;
        let raw_sum = sum + live;
        sum = raw_sum & 268435455;
    }
    std::process::exit((sum & 0xFF) as i32);
}

File: misc/benchmarks/reference/csharp/RedundantStress.cs
/* redundant_stress — Redundant computation / constant propagation benchmark (50M iterations).
   Equivalent to examples/il/benchmarks/redundant_stress.il */
using System;
using System.Runtime.CompilerServices;

class RedundantStress
{
    [MethodImpl(MethodImplOptions.NoInlining)]
    static long Run(long n)
    {
        long sum = 0;
        for (long i = 0; i < n; ++i)
        {
            long k1 = 10 + 20;
            long k2 = k1 * 3;
            long k3 = k2 - 40;

            long a1 = i + 7;
            long a2 = a1 * 3;

            long b1 = i + 7;
            long b2 = b1 * 3;

            long c1 = 100 + 200;
            long c2 = c1 * 2;
            long c3 = c2 - 100;

            long d1 = 5 + 10;
            long d2 = d1 * 5;
            long d3 = d2 - 5;

            long live = a2 + b2 + k3 + c3 + d3;

            long rawSum = sum + live;
            sum = rawSum & 268435455;
        }
        return sum;
    }

    static int Main(string[] args)
    {
        long result = Run(50000000 + args.Length);
        return (int)(result & 0xFF);
    }
}
